src/overlays/actors/ovl_En_Grasshopper/z_en_grasshopper.c

```diff
@@ -15,41 +16,114 @@ void EnGrasshopper_Destroy(Actor* thisx, PlayState* play);
+void EnGrasshopper_RaiseTail(EnGrasshopper* this);
+void EnGrasshopper_LowerTail(EnGrasshopper* this);
+void EnGrasshopper_DecideAction(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupFly(EnGrasshopper* this);
+void EnGrasshopper_Fly(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_RoamInCircles(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupBank(EnGrasshopper* this);
+void EnGrasshopper_Bank(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupBounced(EnGrasshopper* this);
+void EnGrasshopper_Bounced(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupApproachPlayer(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_ApproachPlayer(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupAttack(EnGrasshopper* this);
+void EnGrasshopper_Attack(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupWaitAfterAttack(EnGrasshopper* this);
+void EnGrasshopper_WaitAfterAttack(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_Damaged(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_Dead(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_SetupFall(EnGrasshopper* this);
+void EnGrasshopper_Fall(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_InitializeEffect(EnGrasshopper* this, Vec3f* pos);
+void EnGrasshopper_UpdateEffects(EnGrasshopper* this, PlayState* play);
+void EnGrasshopper_DrawEffects(EnGrasshopper* this, PlayState* play);
+
+typedef enum {
+    /*  0 */ EN_GRASSHOPPER_ACTION_FLY,
+    /*  1 */ EN_GRASSHOPPER_ACTION_ROAM_IN_CIRCLES,
+    /*  2 */ EN_GRASSHOPPER_ACTION_BANK,
+    /*  3 */ EN_GRASSHOPPER_ACTION_BOUNCED,
+    /*  4 */ EN_GRASSHOPPER_ACTION_DECIDE_ACTION,
+    /*  5 */ EN_GRASSHOPPER_ACTION_APPROACH_PLAYER,
+    /*  6 */ EN_GRASSHOPPER_ACTION_ATTACK,
+    /*  7 */ EN_GRASSHOPPER_ACTION_WAIT_AFTER_ATTACK,
+    /*  8 */ EN_GRASSHOPPER_ACTION_DAMAGED,
+    /*  9 */ EN_GRASSHOPPER_ACTION_DEAD,
+    /* 10 */ EN_GRASSHOPPER_ACTION_FALL,
+} EnGrasshopperAction;
+
+typedef enum {
+    /* 0 */ EN_GRASSHOPPER_DECISION_ATTACK,
+    /* 1 */ EN_GRASSHOPPER_DECISION_FLY,
+    /* 2 */ EN_GRASSHOPPER_DECISION_ROAM_IN_CIRCLES, // Never used in the final game
+} EnGrasshopperNextAction;
+
+typedef enum {
+    /* 0 */ EN_GRASSHOPPER_BANK_STATE_BANKING,
+    /* 1 */ EN_GRASSHOPPER_BANK_STATE_DONE,
+} EnGrasshopperBankState;
+
+static s32 sOccupiedIndices[] = {
+    false, false, false, false, false,
+};
+
+static s8 sLimbIndexToShadowBodyPartsIndex[] = {
+    -1, -1, 0, 1, 2, 3, 4, 5, -1, -1, 6, 7, -1, 8, 9, -1, 10, 11, -1, 12, 13, -1, -1, -1,
+};
+
+static s8 sParentBodyParts[] = {
+    -1, -1, -1, 0, 1, 2, 3, -1, -1, -1, -1, 6, -1, -1, 8, -1, -1, 10, -1, -1, 12, -1, -1, -1,
+};
+
+static u8 sShadowSizes[] = {
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+};
+
+typedef enum {
+    /* 0x0 */ EN_GRASSHOPPER_DMGEFF_NO_DAMAGE,  // Deals no damage
```

Comment:
If it has no effect we usually say `IMMUNE`, since damage is generally handled by the actual number.